src/cart/crt_hg.c

```diff
@@ -327,6 +467,85 @@ crt_hg_addr_lookup(struct crt_hg_context *hg_ctx, const char *name,
 	return rc;
 }
 
+int
+crt_hg_remove_addr (hg_class_t *hg_class, hg_addr_t hg_addr)
+{
+        hg_return_t     ret = HG_SUCCESS;
+
+        D_DEBUG(DB_TRACE, "removing addr from class \n" );
+        ret = HG_Addr_set_remove(hg_class, hg_addr);
+        if (ret != HG_SUCCESS) {
+                D_ERROR("HG_Addr_set_remove) failed, hg_ret %d.\n", ret);
+              	D_GOTO(out, ret = -DER_HG);
+        }
+
+        D_DEBUG(DB_TRACE, "freeing addr from class\n");
+        ret = HG_Addr_free(hg_class, hg_addr);
+        if (ret != HG_SUCCESS) {
+                D_ERROR("HG_Addr_free() failed, hg_ret %d.\n", ret);
+              	D_GOTO(out, ret = -DER_HG);
+        }
+
+out:
+        return ret;
+}
+int
+crt_hg_remove_client_id (uint64_t client_id)
+{
+        hg_return_t     	ret = HG_SUCCESS;
```

Comment:
there need to be checks here:
1. if (!is_service()) return -- only supported on clients
2. only for psm2